### PR TITLE
Add Bean Features

### PR DESCRIPTION
**Pull request by ofthedove**
Add features necessary for a bean to the gea3 interface.

Issue: https://github.com/geappliances/applcommon.meta/issues/1514

Detailed changes:
- Update Readme
- Add `forward` command
- Add return value to `send` and `forward`
- Add option `ignoreDestinationAddress`
- Fix a TDD test case

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,3 +9,10 @@ Provides a simple interface for sending and receiving GEA3 serial packets.
 
 ### `tiny_erd_client`
 Provides a simple interface for reading and writing addressable data (ERDs) over a GEA3 serial interface.
+
+## Dev Environment
+
```
Comment (ryanplusplus) [inline README.md]:
```suggestion
```

File: include/tiny_gea3_interface.h [hunk 2]
```diff
@@ -25,7 +25,7 @@ typedef struct {
   i_tiny_gea3_interface_t interface;
 
   tiny_event_t on_receive;
-  tiny_event_subscription_t byte_received_subscfription;
+  tiny_event_subscription_t byte_received_subscription;
```
Comment (ryanplusplus) [inline include/tiny_gea3_interface.h:28]: 👍

File: README.md [hunk 3]
```diff
@@ -9,3 +9,10 @@ Provides a simple interface for sending and receiving GEA3 serial packets.
...
+1. Clone the repo
+2. Install Cpputest with `apt install cpputest` on linux or `brew install cpputest` on mac
+   1. TODO: Add details of mac specific fix here
```
Comment (ryanplusplus) [inline README.md]: Add these lines to `~/.bash_profile` after `eval "$(/opt/homebrew/bin/brew shellenv)"` on macOS:

```
export CPATH="$CPATH:$(brew --prefix)/include"
export LIBRARY_PATH="$LIBRARY_PATH:$(brew --prefix)/lib"
```

**Comment by ryanplusplus** [comment]
Can you bump the version in [`library.json`](https://github.com/geappliances/tiny-gea3-api/blob/main/library.json#L16)? Since this is a breaking change we should bump the major version.

If you have other breaking changes coming, we can add a `develop` branch and accumulate changes there before releasing.

File: README.md [hunk 4]
```diff
@@ -9,3 +9,10 @@ Provides a simple interface for sending and receiving GEA3 serial packets.
 
 ### `tiny_erd_client`
 Provides a simple interface for reading and writing addressable data (ERDs) over a GEA3 serial interface.
+
+## Dev Environment
+
+1. Clone the repo
+2. Install Cpputest with `apt install cpputest` on linux or `brew install cpputest` on mac
```
Comment (cbeeler) [inline README.md]:
```suggestion
2. Install CppUTest with `sudo apt install cpputest` on Ubuntu/Debian or `brew install cpputest` on macOS
```

File: include/tiny_gea3_interface.h (hunk 2 above)
Comment (CeciThomas) [reply include/tiny_gea3_interface.h:28]: `byte_received_subscfriction`. 🤣

**Review by CeciThomas — COMMENTED** [review]
I did a quick review and nothing really stands out other than the misspelling you corrected.  I'll look at it more closely tomorrow if needed.

**Comment by ofthedove** [comment]
> If you have other breaking changes coming, we can add a `develop` branch and accumulate changes there before releasing.

This is a great idea, please do create a develop branch. I know we'll have at least one more significant PR going in, and probably more as we build out the bean.

**Comment by ryanplusplus** [comment]
> > If you have other breaking changes coming, we can add a `develop` branch and accumulate changes there before releasing.
> 
> This is a great idea, please do create a develop branch. I know we'll have at least one more significant PR going in, and probably more as we build out the bean.

Done

**Review by ryanplusplus — APPROVED** [review]

**Pull request merged**